### DISCUSSION TITLE
Prevent WebXR from moving model to fit within screen in AR

### DISCUSSION THREAD

**Discussion by StainlessReality** [google/model-viewer#3939] [Q&A]
Hello!

The problem:

I have a dimensionally large model (6x6x6m for sake of argument) and a small device (Pixel 3a), and when I place it in AR mode (WebXR) it is placed far away from the camera, seemingly in order to ensure the whole model is visible from the start. The design intent is to allow the user to explore the model from up-close, therefore I need the model origin to be kept in the placement point (anchor? AR origin?) upon placement. As it stands the model is placed so far that it gets difficult and fiddly trying to drag the model back towards the user. Sometimes you can't even hit the target area to drag-move the model because it is so small.

I have already tried setting the scale to 0.1 in script, the problem with this is that pinch-scaling in AR then no longer snaps to scale of 1 and instead snaps to scale of 0.1

The experience relies on some UI, model switching and animations inside the AR session, therefore the solution needs to work in WebXR as opposed to SceneViewer (or QuickLook)

Possible solutions, in order of preference:
- Some way to prevent WebXR from trying to keep the model framed within the screen at placement
- Some way to place the model at reduced scale, but still have AR pinch-scale snapping at true 1:1 scale (as opposed to the reduced scale)
- Some way to reset the model position with a button, i.e. snapping the model origin to the point on the floor where the camera was pointing at the time of placement

Any ideas as to how the above solutions could be implemented would be greatly appreciated!

Many thanks,
Jakub

**Comment by elalish** [comment]
I'm pretty sure you're not the first person who has asked for this; we might need some kind of change. I *think* the AR placement is based on where the 3D camera is when AR is activated. Have you tried setting the camera to a small radius (even zero) to put the user in the middle of the model? If this isn't how it works, maybe it should be. In any case, PRs welcome, and especially videos of desired/undesired behavior.
